src/NerdBank.GitVersioning/ReleaseManager.cs

```diff
@@ -280,7 +280,7 @@ private void UpdateVersion(LibGit2Context context, SemanticVersion oldVersion, S
 
         if (!EqualityComparer<SemanticVersion>.Default.Equals(versionOptions.Version, newVersion))
         {
-            if (versionOptions.VersionHeightPosition.HasValue && SemanticVersion.WillVersionChangeResetVersionHeight(versionOptions.Version, newVersion, versionOptions.VersionHeightPosition.Value))
+            if (versionOptions.VersionHeightOffset >= 0 && versionOptions.VersionHeightPosition.HasValue && SemanticVersion.WillVersionChangeResetVersionHeight(versionOptions.Version, newVersion, versionOptions.VersionHeightPosition.Value))
```

Comment:
I know you're doing exactly what I asked for. In retrospect though, I wonder if we're over-shooting here. If someone had `-200` here, that would almost certainly be for a specific reason that no longer applies after a version bump. And it would be invalid after too, because -199 isn't a valid version number.
Let's change this to only skip on *exactly* `-1`. This fits your scenario, your test, and is the only valid negative number (since it will result in an initial version of 0).
```suggestion
            if (versionOptions.VersionHeightOffset != -1 && versionOptions.VersionHeightPosition.HasValue && SemanticVersion.WillVersionChangeResetVersionHeight(versionOptions.Version, newVersion, versionOptions.VersionHeightPosition.Value))
```

Reply:
Thanks for your reply! Yes, `-1` value looks to be the only valid case.

Reply:
Fair point, @KalleOlaviNiemitalo. We can adjust this further if we get feedback from users that use it. We might require a CLI switch to prevent resetting the value.